### PR TITLE
Slack integration

### PR DESCRIPTION
**Pull request by mariusandra**
Hi, this is a very basic and naïve implementation of #105. More details in the issue.

### CODE REVIEW

File: frontend/src/scenes/actions/ActionEdit.js [hunk 1]
```diff
@@ -187,30 +157,33 @@ export class ActionEdit extends Component {
...
+                        <div style={{ marginTop: 20, marginBottom: 15 }}>
+                            <label>
+                                <input
+                                    type="checkbox"
+                                    onChange={e => {
+                                        this.setState({ action: { ...action, post_to_slack: e.target.checked } })
+                                    }}
+                                    checked={action.post_to_slack}
+                                />
+                                &nbsp;Post to Slack when this action is triggered
```
Comment (timgl) [inline frontend/src/scenes/actions/ActionEdit.js:170]: Could we have a `<small>configure</small>` link here to /setup (maybe even with ref) and show it even if the team doesn't have Slack configured? Maybe greyed out (opacity 0.7 or something) and input disabled

**Review by timgl — APPROVED** [review]

File: posthog/models.py [hunk 2]
```diff
@@ -222,8 +224,12 @@ def create(self, *args: Any, **kwargs: Any):
             for action in event.actions:
                 relations.append(action.events.through(action_id=action.pk, event=event))
             Action.events.through.objects.bulk_create(relations, ignore_conflicts=True)
-            return event
+            for action in event.actions:
+                if action.post_to_slack and action.team and action.team.slack_incoming_webhook:
+                    values = {'text': 'Action "{}" triggered!'.format(action.name)}
+                    requests.post(action.team.slack_incoming_webhook, verify=False, json=values)
```
Comment (mariusandra) [inline posthog/models.py:230]: With the latest rebase, I had to make some funky changes here (two loops)
However this is a WIP anyway, as we'll move this to workers soon enough.

File: frontend/src/scenes/actions/ActionEdit.js (hunk 1 above)
Comment (mariusandra) [reply frontend/src/scenes/actions/ActionEdit.js]: done

File: posthog/api/user.py [hunk 3]
```diff
@@ -33,7 +34,8 @@ def user(request):
             'app_urls': team.app_urls,
             'api_token': team.api_token,
             'signup_token': team.signup_token,
-            'opt_out_capture': team.opt_out_capture
+            'opt_out_capture': team.opt_out_capture,
+            'slack_incoming_webhook': team.slack_incoming_webhook
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply posthog/api/user.py:38]: done

File: frontend/src/scenes/setup/SlackIntegration.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+export function SlackIntegration() {
+    const { user } = useValues(userLogic)
+    const { isSaved } = useValues(logic)
+    const { userUpdateRequest } = useActions(userLogic)
+    const { slack_incoming_webhook: slackIncomingWebhook } = user.team
+    const [editedWebhook, setEditedWebhook] = useState(slackIncomingWebhook || '')
+
+    return (
+        <div>
+            <label>Enter your slack webhook URL here</label>
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/setup/SlackIntegration.js]: done

File: posthog/models.py (hunk 2 above)
Comment (timgl) [reply posthog/models.py:230]: Would you mind moving this up into the property above so it is one loop? The `event.actions` is a really nasty SQL query and in this case we're running it twice.

**Comment by mariusandra** [comment]
Hi, this PR should be ready to be merged. Just how should we test this? I added a few simple tests I could adapt that check editing some params..., but the entire process from registering the right slack webhook URL in /setup to creating an event that makes a worker call that webhook URL is currently untested.

**Comment by timgl** [comment]
@Tannergoods Would be cool if you could test this end to end (so create a slack app etc) [see here for instructions](https://github.com/PostHog/docs/pull/6/files?short_path=28737ae#diff-28737ae88ed88f9e306a72bc4f056b65)

**Comment by Tannergoods** [comment]
The "Slack Integration" isn't in setup yet? 
Posthog

**Comment by mariusandra** [comment]
@Tannergoods It should be right below the "api keys" and above the "invite your team" blocks... (if you're on the review app for this branch of course)

**Comment by timgl** [comment]
The review app was deleted due to inactivity, but it's re-deploying now

**Comment by Tannergoods** [comment]
Works great up to instruction no.4 where I can't save the action whilst also ticking "Post to Slack..."

Posthog

It doesn't work if you try and save either before or after selecting the element using front end toolbar but pageviews work fine as does custom events

Slack_____slackupdates___PostHog

**Comment by mariusandra** [comment]
Hi @Tannergoods that is fixed. While I was working on this feature, there was work in another branch that disabled the "save" button if nothing was changed on the page (#497). I integrated this new requirement to the slack checkbox as well.

This `ActionEdit.js` should actually be refactored to use Kea...

**Comment by Tannergoods** [comment]
That's great @mariusandra everything works in the way I expect. Great job!

**Comment by timgl** [comment]
Merged!

**Pull request merged**